### PR TITLE
[MBL-722] Full MockOptimizely Removal

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Cleans up and removes deprecated optimizely code, updates library tests, and updates/re-records snapshot tests (this is why there are so many file changes)

# 🤔 Why

Remote Config is fully integrated now so the optimizely code is no longer being used and can be cleaned up.

# 🛠 How

1. Remove each optimizely file, follow the build errors to remove unused code, triple check that the features that relied on optimizely flag values default to true (except comment flagging which should be false).
2. Update existing tests and remove irrelevant ones in Library target
3. Update snapshot tests and re-record the references if needed in Kickstarter-iOS target.

# ✅ Acceptance criteria

Please review this commit by commit. There were a lot of snapshots updated.

- [x] The should behave entirely as before this change. 
- [x] The only feature flags we have not are for the Consent Management Dialog and Facebook Interstitial features and both are off and should not be shown in the app

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewControllerTests.swift [hunk 1]
```diff
@@ -151,77 +151,12 @@ internal final class ProjectPageViewControllerTests: TestCase {
...
-        assertSnapshot(
-          matching: parent.view,
-          as: .image(perceptualPrecision: 0.98),
-          named: "lang_\(language)_device_\(device)"
-        )
-      }
-    }
-  }
-
-  func testLoggedIn_Backer_LiveProject_NotShowingCampaign_Success() {
```
[the remark this answers is not in the document]
Comment (msadoon) [reply Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewControllerTests.swift:186]: Also no changes should be introduced for `ProjectPageViewControllerConversionTests` under `Snapshots`, there were 0 changes to that test file.

**Review by msadoon — COMMENTED** [review]
Ok, so there's a lot here that is correct. But I really want you to focus on the changes that are needed here and _understand_ _why_ they need be reverted.

Please put together a PR that includes all these changes. I would like it if I only have to review in _once_ so all these comments are considered and changes made. Ideally we should merge that PR after a single pass.

Please ask me any questions you have or if anything is unclear.

File: Library/ViewModels/CommentCellViewModelTests.swift [hunk 2]
```diff
@@ -95,54 +90,17 @@ internal final class CommentCellViewModelTests: TestCase {
...
-
-      self.bottomRowStackViewIsHidden
-        .assertValue(
-          true,
-          "The feature flags are false, therefore the stack view is hidden."
-        )
-    }
-  }
-
-  func testOutputs_bottomRowStackViewIsHidden_LoggedOut_FeatureFlag_True() {
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Library/ViewModels/CommentCellViewModelTests.swift:118]: Now that the comment flagging feature is always `false`, the `bottomRowStackView` will always be hidden and therefore these test cases will never pass. 
There are already test cases that assert the `true` cases on lines 79, 93, 106, and 116.

File: Library/ViewModels/CommentCellViewModelTests.swift [hunk 3]
```diff
@@ -95,54 +90,17 @@ internal final class CommentCellViewModelTests: TestCase {
...
-
-      self.bottomRowStackViewIsHidden
-        .assertValue(
-          false,
-          "The comment flagging feature is enabled, therefore the stack view is not hidden."
-        )
-    }
-  }
-
-  func testOutputs_bottomRowStackViewIsHidden_IsNotBacking_FeatureFlag_True() {
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Library/ViewModels/CommentCellViewModelTests.swift:133]: see my reply in the comment above

File: Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewControllerTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (msadoon) [reply Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewControllerTests.swift:186]: I'll double check of the test screenshots and ensure that the link is replaced by the tab. Just add back in the deleted screenshots in the new pr

**Pull request merged**